fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/PickupItemCallback.java

```diff
@@ -0,0 +1,34 @@
+package net.fabricmc.fabric.api.event.player;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.ActionResult;
+import net.minecraft.world.World;
+
+/**
+ * Callback for picking up an item.
+ * Is hooked in before the item is picked up.
+ * <p>
+ * Upon return:
+ * - SUCCESS cancels further processing and picks up the item.
+ * - PASS falls back to further processing.
+ * - FAIL cancels further processing and does not pick up the item.
+ */
+public interface PickupItemCallback {
+	public static final Event<PickupItemCallback> EVENT = EventFactory.createArrayBacked(PickupItemCallback.class,
+		(listeners) -> (player, world, entity, amount) -> {
+			for (PickupItemCallback event : listeners) {
+				ActionResult result = event.interact(player, world, entity, amount);
+				if (result != ActionResult.PASS) {
+					return result;
+				}
+			}
+
+			return ActionResult.PASS;
+		}
+	);
+
+	ActionResult interact(PlayerEntity player, World world, Entity pickupEntity, int amount);
```

Comment:
Why is `pickupEntity` not an `ItemEntity` ? Are there any situation where this event could be fired on something else ? If the change is made, `amount` becomes redundant too.

Reply:
The event originally supported both xp/item entities, so I forgot to change the interact parameters. Fixed!